modulus/models/module.py

```diff
@@ -65,55 +105,256 @@ def debug(self):
         # TODO: set up debug log
         # fh = logging.FileHandler(f'modulus-core-{self.meta.name}.log')
 
-    def save(self, file_name: Union[str, None] = None) -> None:
+    def save(
+        self, file_name: Union[str, None] = None, save_git_hash: bool = False
+    ) -> None:
         """Simple utility for saving just the model
 
         Parameters
         ----------
         file_name : Union[str,None], optional
             File name to save model weight to. When none is provide it will default to
             the model's name set in the meta data, by default None
+        save_git_hash : bool, optional
+            Whether to save the git hash of the current commit, by default False
 
         Raises
         ------
         IOError
             If file_name provided has a parent path that does not exist
         """
+
+        if file_name is not None and not file_name.endswith(self._file_extension):
+            raise ValueError(
+                f"File name must end with {self._file_extension} extension"
+            )
+
+        directory = Path(file_name)
+        if not directory.parents[0].is_dir():
+            raise IOError(
+                f"Model checkpoint parent directory {directory.parents[0]} not found"
+            )
+        if not directory.is_dir():
+            os.makedirs(directory)
+
+        torch.save(self.state_dict(), directory.joinpath("model.pt"))
+
+        with open(directory.joinpath("args.json"), "w") as f:
+            json.dump(self._args, f)
+            if file_name is None:
+                file_name = self.meta.name + ".pt"
+
+        # Save the modulus version and git hash (if available)
+        version_info = {"modulus_version": modulus.__version__}
+
+        if save_git_hash:
+            import git
+
+            repo = git.Repo(search_parent_directories=True)
+            try:
+                version_info["git_hash"] = repo.head.object.hexsha
+            except git.InvalidGitRepositoryError:
+                version_info["git_hash"] = None
+
+        with open(directory.joinpath("version.json"), "w") as f:
+            json.dump(version_info, f)
+
```

Comment:
Would we ever want a util that converts from mdlus to just a pytorch checkpoint?

Reply:
I am open to this but I can't really think of any reason this would be used. We are inheriting from nn.Module already and being careful to keep all the needed nn.Module methods. Someone should be able to use a modulus.Module exactly like a pytorch model and have no issues. I can see some problems with type checking but I don't know, that kinda feels like their problem as all models will be inherited from nn.Module anyways.

Reply:
Yeah, its fine without. Just wanted to throw it out there and see your throughts. Not sure I exactly see the use case right now since the point of these packages is to use the Modulus utils. Resolving.